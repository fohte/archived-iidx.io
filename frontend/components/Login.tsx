import { auth } from '../lib/firebaseApp'

const handleLogin = async () => {
  try {
    await auth.signInWithEmailAndPassword(
      window.prompt('email') || '',
      window.prompt('password') || '',
    )
  } catch (e) {
    console.error(e)
  }
}

export default () => <button onClick={handleLogin}>login</button>
