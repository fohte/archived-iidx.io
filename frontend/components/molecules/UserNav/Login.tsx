import { Button } from 'antd'
import Link from 'next/link'
import * as React from 'react'

const Login: React.SFC = () => (
  <Link href="/login" prefetch>
    <Button>Login</Button>
  </Link>
)

export default Login
