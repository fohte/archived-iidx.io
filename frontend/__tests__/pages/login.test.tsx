import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Login from '@pages/login'

describe('/login', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Login />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
