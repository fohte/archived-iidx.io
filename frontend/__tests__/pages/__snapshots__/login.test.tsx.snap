// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/login renders correctly 1`] = `
Array [
  <div
    className="ui inverted vertical segment"
  >
    <div
      className="ui inverted secondary menu"
    >
      <div
        className="ui container"
      >
        <a
          className="item"
          onClick={[Function]}
        >
          <span>
            iidx.io
          </span>
        </a>
        <div
          className="right menu"
        >
          <div
            className="ui tiny active inline loader"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui container"
    >
      <div
        className="ui center aligned middle aligned grid"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="column"
          style={
            Object {
              "maxWidth": 450,
            }
          }
        >
          <h2
            className="ui teal center aligned header"
          >
            Log-in to your account
          </h2>
          <form
            className="ui large form"
            onSubmit={[Function]}
          >
            <div
              className="ui stacked left aligned segment"
            >
              <div
                className="field"
              >
                <div
                  className="ui fluid left icon input"
                >
                  <input
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="E-mail address"
                    type="email"
                    value=""
                  />
                  <i
                    aria-hidden="true"
                    className="user icon"
                  />
                </div>
              </div>
              <div
                className="field"
              >
                <div
                  className="ui fluid left icon input"
                >
                  <input
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <i
                    aria-hidden="true"
                    className="lock icon"
                  />
                </div>
              </div>
              <button
                className="ui teal large fluid disabled button"
                disabled={true}
                onClick={[Function]}
                tabIndex={-1}
                type="submit"
              >
                Log in
              </button>
            </div>
          </form>
          <div
            className="ui message"
          >
            New to us?
             
            <a
              href="/signup"
              onClick={[Function]}
            >
              Sign up.
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ui vertical segment"
  >
    <div
      className="ui center aligned container"
    >
      <i
        aria-hidden="true"
        className="copyright outline icon"
      />
       
      2018
       iidx.io |
       
      test
    </div>
  </div>,
]
`;
